File: algorytm/Python/algorytm.py
def algorytm(array):
	# Keep track of smallest in
	smallest = 1


	# Sort the array
	array.sort()

	if array[-1] <= 0:
		return smallest


	for i in array:
		if i > smallest:
			return smallest
		elif i == smallest:
			smallest+=1

	return smallest


test1 = [3, 5, -2, 2, 1, 8, 15] #4
test2 = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] #2
test3 = [-5, -3, 0, 0, -2] #1

print(algorytm(test1))
print(algorytm(test2))
print(algorytm(test3))

File: algorytm/Java/algorytm.java
import java.util.Arrays;

public class Main{

	public static void main(Stringp[] args) {
		int[] test1 = {3, 5, -2, 2, 1, 8, 15};

		System.out.println(algorytm(test1));
	}

	public int algorytm(int[] array) {
		int smallest = 1;

		Arrays.sort(array);

		if (array[array.length-1] <= smallest) {
			return smallest;
		}

		for (int i = 0; i < array.length; i++) {
			if (array[i] > smallest) {
				return smallest;
			}
			else if (array[i] == smallest) {
				smallest++;
			}
		}

		return smallest;
	}
}